dask_expr/_merge.py

```diff
@@ -284,12 +284,14 @@ def _lower(self):
                             left,
                             shuffle_left_on,
                             npartitions_out=left.npartitions,
+                            backend="tasks",
```

Comment:
Why does this rely on `tasks`?

Reply:
To keep the hashing consistent, not ideal but everything else loses data at the moment

Reply:
Cleaned this up now after we removed the bug in dask/dask